src/tests/Microsoft.PowerFx.Core.Tests/CheckResultTests.cs

```diff
@@ -262,70 +262,36 @@ public void BindingChangeSymbols()
         }
 
         [Theory]
-        [InlineData("\"test string\"", false, true)]
-        [InlineData("\"test string\"", true, true)]
-        [InlineData("12", false, false)]       
-        [InlineData("12", true, true)]
-        [InlineData("{a:12, b:15}", true, false)]
-        [InlineData("{a:12, b:15}", false, false)]
-        public void CheckResultExpectedReturnValueString(string inputExpr, bool allowCoerceTo, bool isSuccess)
-        {
-            var check = new CheckResult(new Engine())
-                .SetText(inputExpr)
-                .SetBindingInfo()
-                .SetExpectedReturnValue(FormulaType.String, allowCoerceTo);
-
-            if (isSuccess)
-            {
-                Assert.True(check.IsSuccess);
-            }
-            else
-            {
-                var errors = check.ApplyErrors();
-
-                Assert.False(check.IsSuccess);
-                Assert.Single(errors);
-                var error = errors.First();
-                Assert.Contains("The type of this expression does not match the expected type 'Text'", error.ToString());
-            }
+        [InlineData("\"test string\"", false, true, "")]
+        [InlineData("\"test string\"", true, true, "")]
+        [InlineData("12", false, false, "The type of this expression does not match the expected type 'Text'")]       
+        [InlineData("12", true, true, "")]
+        [InlineData("{a:12, b:15}", true, false, "The type of this expression does not match the expected type 'Text'")]
+        [InlineData("{a:12, b:15}", false, false, "The type of this expression does not match the expected type 'Text'")]
+        public void CheckResultExpectedReturnValueString(string inputExpr, bool allowCoerceTo, bool isSuccess, string errorMsg)
+        {
+            CheckResultExpectedReturnValue(inputExpr, allowCoerceTo, isSuccess, errorMsg, FormulaType.String);
         }
 
         [Theory]
         [InlineData("12", false, true, "")]
         [InlineData("12", true, true, "")]
-        [InlineData("\"test string\"", true, false, "The method or operation is not implemented")]
-        [InlineData("\"test string\"", false, false, "The type of this expression does not match the expected type 'Number'")]
```

Comment:
we now allow numeric return type so we allow string convert to number. Remove these test to have clear test for right string to number or non support type (record) to number

Reply:
Can we keep the test and change the `isSuccess` value?
I wouldn't want to loose the test coverage from existing tests...

Reply:
This test is wrong test anyway; [InlineData("\"test string\"", true, false, "The method or operation is not implemented")] where this string cannot coerce to number but I set allowtocoerceto is true. This test need to remove. It passed last time because we did not support it. But now we support number and it will show the failure. I don't think we should keep this test.

Reply:
I think we should keep the test and change it to below, 
Here's why,
First 2 test checks for number, which would always return number type, since input is number type.
Last 2 test checks for record type, which can never be coerced to number type.
The below 2 test would check for something that isn't number type and can be coerced to number type.
`[InlineData("\"test string\"", true, true, "")]`
`[InlineData("\"test string\"", false, false, "The type of this expression does not match the expected type 'Number'")]`

Reply:
i will keep it then. thanks